src/java.compiler/share/classes/javax/lang/model/element/ElementKind.java

```diff
@@ -121,7 +123,14 @@ public enum ElementKind {
      * A binding variable in a pattern.
      * @since 16
      */
-    BINDING_VARIABLE;
+    BINDING_VARIABLE,
+
+    /**
+     * A local component variable in a derived record creation expression.
+     * @since 23
+     */
+    @PreviewFeature(feature=PreviewFeature.Feature.DERIVED_RECORD_CREATION, reflective=true)
+    COMPONENT_LOCAL_VARIABLE;
 
     // Maintenance note: check if the default implementation of
```

Comment:
From a quick look, I don't think Elements.getOutermostTypeElement needs updating for COMPONENT_LOCAL_VARIABLE, but it would be good to add a test case.

Reply:
Tests added: https://github.com/openjdk/jdk/pull/18509/commits/e09306885e5c6cd2aba7779025fc7efcb8991e8e

Thanks!